### PR TITLE
843 fix json warning bug

### LINKED ISSUE #843: JSONReader.warnings sometimes returns string type
In **nexus_constructor/main_window.py** we have following code:

`            if reader.warnings:
                show_warning_dialog(
                    "\n".join(
                        (json_warning.message for json_warning in reader.warnings)
                    ),
                    "Warnings encountered loading JSON",
                    parent=self,
                )
`

However, the warnings attribute in reader can sometimes contain a pure string element. Accessing the message attribute from a `str` type is of course not valid and an `AttributeError` is thrown. This will lead to JSON files not being loaded correctly.

Fix this issue by making sure that JSONReader always contains appropriate object types in `reader.warnings`, containing a message attribute. Appropriate class can be found in **nexus_constructor/json/json_warnings.py**.

### PR DESCRIPTION
**Pull request by kmurica**
Closes #843

### CODE REVIEW

**Comment by kmurica** [comment]
@ebadkamil I think this is ready for review.

File: nexus_constructor/json/json_warnings.py [hunk 1]
```diff
@@ -30,5 +40,88 @@ class NXClassAttributeMissing:
...
+    InvalidTransformation,
+    TransformDependencyMissing,
+    NameFieldMissing,
+    NXClassAttributeMissing,
 ]
+
+_T = TypeVar("_T", JsonWarning, "JsonWarningsContainer")
+
+
+class JsonWarningsContainer(List[JsonWarning]):
```
Comment (ebadkamil) [inline nexus_constructor/json/json_warnings.py:54]: Just for curiosity, Why `List[JsonWarning] `and not just `List`?

Comment (kmurica) [reply nexus_constructor/json/json_warnings.py:54]: As we discussed in chat, in principle it is OK to use List but it is mostly to be super clear that it is a JsonWarning container. :)

**Review by ebadkamil — COMMENTED** [review]
LGTM. In case you need some more opinion, feel free to add others as well.

**Review by ebadkamil — APPROVED** [review]

**Pull request merged**